lib/string_basics.dart

```diff
@@ -200,6 +200,32 @@ extension StringBasics on String {
     }
     return stringBuffer.toString();
   }
+
+  /// Returns a truncated version of the string.
+  ///
+  /// Example:
+  /// ```dart
+  /// final sentence = 'The quick brown fox jumps over the lazy dog';
+  /// final truncated = sentence.limit(20); // 'The quick brown fox...'
+  /// ```
+  ///
+  /// The [limit] is the truncated length of the string.
+  /// The [end] is the suffix or ending string of the truncated string.
+  /// The [trim] is whether or not to trim the spaces of the truncated string
+  /// before appending the ending string.
+  String limit(int limit, {String end = '...', bool trim = true}) {
```

Comment:
I think I'd be inclined to just drop the default argument here altogether. To my mind, if I call `"abcd".truncate(3)` with no special arguments, I'd expect "abc" rather than "..." or "abc…". I also think using the unicode character could potentially create confusion, where a user who doesn't understand that it's a single character that looks like 3 will think that the method is exhibiting a bug or unexpected behavior, whereas if we just let them provide the substitution themselves, it will always be clear what's going on.

Reply:
I updated the `substitution` parameter to default to empty string.